### PR TITLE
DOMA-2241 Link BillingRecipient and BillingReceipt

### PR DESCRIPTION
**Pull request by toplenboren**
Adds new relation field to `billingReceipt` (rel to `BillingRecipient`) 

Also adds `billingReceipt.onUpdate` hook, which creates `BillingRecipient:rel` using data from `recipient:JSON` field

### CODE REVIEW

File: apps/condo/domains/billing/schema/BillingReceipt.js [hunk 1]
```diff
@@ -87,6 +102,59 @@ const BillingReceipt = new GQLListSchema('BillingReceipt', {
...
+                if (!recipient) {
+                    recipient = get(existingItem, 'recipient')
+                }
+                contextId = get(resolvedData, 'context')
+                if (!contextId) {
+                    contextId = get(existingItem, 'context')
+                }
+            }
+
+            let receiverId
```
Comment (SavelevMatthew) [inline apps/condo/domains/billing/schema/BillingReceipt.js:134]: getOne is called on every api call :((((
May be we should decrease this amount by skipping some cases?
For example, if receiver is already specified...

How about `if(!existingItem.receiver || resolvedData.recepient)`

File: apps/condo/domains/billing/gql.js [hunk 2]
```diff
@@ -40,7 +40,7 @@ const BILLING_RECEIPT_TO_PAY_DETAILS_FIELDS = 'toPayDetails { charge formula bal
 const BILLING_RECEIPT_SERVICE_TO_PAY_DETAILS_FIELDS = BILLING_RECEIPT_TO_PAY_DETAILS_FIELDS.replace('}', 'volume tariff measure }')
 const BILLING_RECEIPT_SERVICE_FIELDS = `services { id name toPay ${BILLING_RECEIPT_SERVICE_TO_PAY_DETAILS_FIELDS} }`
 const BILLING_RECEIPT_RECIPIENT_FIELDS = 'recipient { tin iec bic bankAccount }'
-const BILLING_RECEIPT_FIELDS = `{ context ${BILLING_INTEGRATION_ORGANIZATION_CONTEXT_FIELDS} importId property { id, address } account { id, number, unitName } period raw toPay printableNumber ${BILLING_RECEIPT_TO_PAY_DETAILS_FIELDS} ${BILLING_RECEIPT_SERVICE_FIELDS} ${BILLING_RECEIPT_RECIPIENT_FIELDS} ${COMMON_FIELDS} }`
+const BILLING_RECEIPT_FIELDS = `{ context ${BILLING_INTEGRATION_ORGANIZATION_CONTEXT_FIELDS} importId property { id, address } account { id, number, unitName } period raw toPay printableNumber ${BILLING_RECEIPT_TO_PAY_DETAILS_FIELDS} ${BILLING_RECEIPT_SERVICE_FIELDS} receiver { id tin iec bic bankAccount } ${BILLING_RECEIPT_RECIPIENT_FIELDS} ${COMMON_FIELDS} }`
```
[the remark this answers is not in the document]
Comment (SavelevMatthew) [reply apps/condo/domains/billing/gql.js:43]: Oh...I guess it's for tests...then it's ok....resolved

File: apps/condo/bin/fix-billing-receipts.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+        const receipt = allBillingReceipts[i]
+
+        const receiptId = get(receipt, 'id')
+        const recipient = get(receipt, 'recipient')
+        const contextId = get(receipt, ['context', 'id'])
+
+        let receiverId
+        const sameRecipient = await BillingRecipient.getOne(adminContext, {
+            tin: get(recipient, 'tin'),
```
[the remark this answers is not in the document]
Comment (toplenboren) [reply apps/condo/bin/fix-billing-receipts.js:47]: This field is required, so it wont render as undefined. But i've implemented your suggestion to `name` field

File: apps/condo/domains/billing/schema/BillingReceipt.js (hunk 1 above)
Comment (toplenboren) [reply apps/condo/domains/billing/schema/BillingReceipt.js:134]: Done!

File: apps/condo/domains/billing/schema/BillingReceipt.js [hunk 4]
```diff
@@ -87,6 +102,59 @@ const BillingReceipt = new GQLListSchema('BillingReceipt', {
...
+            if (operation === 'update') {
+                recipient = get(resolvedData, 'recipient')
+                if (!recipient) {
+                    recipient = get(existingItem, 'recipient')
+                }
+                contextId = get(resolvedData, 'context')
+                if (!contextId) {
+                    contextId = get(existingItem, 'context')
+                }
+            }
```
[the remark this answers is not in the document]
Comment (toplenboren) [reply apps/condo/domains/billing/schema/BillingReceipt.js]: Done!

**Pull request merged**